### DISCUSSION TITLE
X3 Pro Gen2 only 4 Entities available?

### DISCUSSION THREAD

**Discussion by BolleY2K** [wills106/homeassistant-solax-modbus#519] [Q&A]
First thank you very much for this very useful piece of software. It works perfectly fine out of the box with my Solax Hybrid Gen4, but using it with my other inverter, a X3 Pro Gen2, yields only 4 not very useful entities. I know there is much more available from:

curl -d "optType=ReadRealTimeData&pwd=REG_NO" -X POST http://192.xxx.xxx.xxx

I am polling both inverters over the PocketWifi with Modbus TCP. Any advice?

Sorry if I forgot any important information, I am very new to Home Assistant and Modbus also.

**Comment by wills106** [comment]
Does it return an error in the log:
"unrecognized inverter type - serial number : {your_serial_number_here}"
Or just "cannot find serial number, even not for MIC"

**Reply by wills106** [reply]
Also I have moved it to the discussions section as it isn't an issue with the integration.

**Reply by BolleY2K** [reply]
Appreciated. Sorry, I am also new to Github but will happily adapt and learn.

The relevant errors shown in the log are:

unrecognized inverter type - serial number : unknown
12:54:28 – ...
SolaXPro: cannot find serial number, even not for MIC
12:54:28 – ...
SolaXPro: reading serial number from address 0x300 failed; other address may succeed
12:54:28 – ...
SolaXPro: reading serial number from address 0x0 failed; other address may succeed
12:54:23 – ...

**Reply by wills106** [reply]
As it's saying unknown you don't have a working Modbus connection.

I would first double check you have the latest [Firmware on the Pocket WiFi](https://github.com/wills106/homeassistant-solax-modbus/edit/main/README.md#pocket-wifi-30-with-firmware-v300403-and-above-is-only-officially-supported-solax-only-mention-gen4-hybrid-other-inverter-may-work)

[But I don't think it provides a Modbus connection on the X3 Mic / Pro](https://community.home-assistant.io/t/universal-solar-inverter-over-modbus-rs485-tcp-custom-component-growatt-sofar-solax-solis/140143/1226)

**Reply by BolleY2K** [reply]
You might be right, since the Curl query listed above uses the API, not Modbus. I tested with another tool for Modbus TVP with the same result - Hybrid works, Pro does not.

Thank you for your time and feedback!

**Reply by wills106** [reply]
If you still want to connect your Inverter you can have a look at the [listed adaptors here](https://github.com/wills106/homeassistant-solax-modbus/wiki/Compatible-RS485-Adaptors)

**Reply by MHBraun** [reply]
Same problem here after update to 
Home Assistant 2023.9.2
Supervisor 2023.09.2
Operating System 10.5
Frontend 20230911.0 - latest
SolaX 2023.09.02
QCells Serial starting with H34B15 with firmware 3.001.02
Before the upgrade I had more than 240 entities, now only 4

**Reply by wills106** [reply]
Have you checked the Releases page? There has been an update that should fix your problem https://github.com/wills106/homeassistant-solax-modbus/releases

**Comment by MHBraun** [comment]
Sorry, no change. Still the same four entities after the upgrade.


________________________________
